### ISSUE TITLE
Bug: PacketItemNBTToServer can be used to change NBT for non-RFTools items

### ISSUE THREAD

**Issue by calvinbalke13** [McJty/RFToolsControl#4]
PacketItemNBTToServer should have some validation on the server side to make sure that the item is actually an RFTools item.  Without that, it creates a lot of vulnerabilities with other modded items.

**Comment by McJty** [comment]
How can it create vulnerabilities? You need a mod to use this packet and if you have a mod then you can do it yourselves

**Comment by calvinbalke13** [comment]
I created an example Proof of Concept exploit for the combination of Storage Drawers and RFToolsControl which allows spawning any item in the game, even Creative Mode items.  It will work without privileges on the server, and the exploit I created is client-side only.  Basically, some mods like Storage Drawers put tile entity data into the tag that PacketItemNBTToServer allows changing.  In the case of Storage Drawers, the NBT data can contain ItemStacks in the inventory.  This effects modded items that don't use RFToolsControl.  There can also be problems with RFTools and any other mod that stores important data in the NBT compound tag.  Since there isn't any normal way to edit NBT data on a server using the client, this mod/packet can cause serious problems if it is misused.
